### ISSUE TITLE
Flatlist crashes

### ISSUE THREAD

**Issue by billtlee** [facebook/react-native#26138]
I am trying to display a flatlist of transaction history from etherscan and the flatlist will only display 10 items before it crashes.

React Native version:
package.json dependencies
    "ethers": "^4.0.33",
    "expo": "^34.0.1",
    "expo-cli": "3.0.10",
    "expo-constants": "~6.0.0",
    "expo-secure-store": "~6.0.0",
    "global": "^4.4.0",
    "react": "16.8.3",
    "react-dom": "^16.8.6",
    "react-native": "https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz",

## Steps To Reproduce

```
  const [walletHistory, setWalletHistory] = useState([]);
  const getWalletHistory = async () => {
    try {
      const etherscanProvider = new ethers.providers.EtherscanProvider(
        (network = 'rinkeby')
      );

      const history = await etherscanProvider.getHistory(wallet.address);
// you can use 0x93D06aBf524bE323a812BF0426A22bF48C04f8db, it's my testing account
// it has about 100 transactions
      history.map(item => {
        const {
          from,
          to,
          hash,
        } = item;

        setWalletHistory(oldarray => [
          ...oldarray,
          {
            from,
            to,
            hash,
            operation:
              from.toLowerCase() === wallet.address.toLowerCase() ? 'out' : 'in'
          }
        ]);
      });
    } catch (e) {}
  };

  const renderItem = ({ item }) => {
    const title = item.hash;
    const leftIcon =
      item.operation === 'in'
        ? { name: 'arrow-down', type: 'entypo', color: 'green' }
        : { name: 'arrow-up', type: 'entypo', color: 'red' };
    return (
      <TouchableOpacity
        onPress={() =>
          navigation.navigate('TransactionDetail', {
            transaction: item
          })
        }
      >
        <ListItem
          chevron
          title={title}
          titleStyle={styles.link}
          leftIcon={leftIcon}
        />
      </TouchableOpacity>
    );
  };

  return (
          <FlatList
            style={{ width: '95%' }}
            ListEmptyComponent={<Text>No Transactions</Text>}
            data={walletHistory}
            keyExtractor={item => item.hash}
            renderItem={renderItem}
          />
  )

```

**Comment by AndreiCalazans** [comment]
This is hard to help you with. Add a Stack Trace. Needless to say though that there is nothing wrong with FlatLists. You can see my snack here rendering a large list with 10+ items loaded after the first render.  https://snack.expo.io/@andreixoc/thoughtful-truffle

Ideally, this should not be an issue in React Native repo but a StackOverflow question.

**Comment by billtlee** [comment]
Hi @AndreiCalazans,

Thanks for the response!  I did a little more digging and found that like you said it's probably not flatlist.  Unfortunately, the crash is manifesting itself while I am trying to display the flatlist.  I have tried to run it on emulators as well as physical devices.  It appears that the crash only happens on my Android physical device.  It doesn't crash when I run it on iOS physical device, iOS emulator, or android emulator.  So, there is a problem somewhere and I don't know where to report it.  When it crashes, the app just freezes with no stack trace information.  Do you know where I should file this?

**Comment by pedrosobral** [comment]
Hi @billtlee!

I was able to run your code on my Android device with no issues, but I had to replace the `ListItem` to a plain `<Text>` because I don't have access to.

I see what may be a performance issue on code that you are calling `setWalletHistory ` over a map, which is not a good thing.

Try to map it first and then call `setWalletHistory` at the end (see code below). I don't' know, it may help you or it's something related to the `ListItem` or memory issues with your physical Android device.

```javascript
const map = history.map(({from, to, hash}) => ({
  from,
  to,
  hash,
  operation:
    from.toLowerCase() === '0x93D06aBf524bE323a812BF0426A22bF48C04f8db'
      ? 'out'
      : 'in',
}));

setWalletHistory(map);
```

Running your code: https://drive.google.com/open?id=1k1dUvbSslaOpwA4QQQQ6z6EpVRAxBzyw
